Assets/_BForBoss/_Weapons/Scripts/Weapons/MeleeWeapon.cs

```diff
@@ -30,26 +30,18 @@ public void DecrementCooldown(float deltaTime)
             _currentCooldown -= _currentCooldown > 0 ? deltaTime : _currentCooldown;
         }
 
-        public bool TryAttackMany(Vector3 playerPosition, Vector3 playerForwardDirection)
+        public int? TryAttackMany(Vector3 playerPosition, Vector3 playerForwardDirection)
         {
-            if (!CanMelee)
-                return false;
-            _currentCooldown += _meleeProperties.AttackCoolDown;
-            
-            _hits = _meleeProperties.OverlapCapsule(playerPosition, playerForwardDirection, TagsAndLayers.Layers.PlayerMask, ref _enemyBuffer);
-            return true;
+            return TryAttack(playerPosition, playerForwardDirection);
         }
         
-        public bool TryAttackOne(Vector3 playerPosition, Vector3 playerForwardDirection)
+        public int? TryAttackOne(Vector3 playerPosition, Vector3 playerForwardDirection)
         {
-            if (!CanMelee)
-                return false;
-            _currentCooldown += _meleeProperties.AttackCoolDown;
-            
-            _hits = _meleeProperties.OverlapCapsule(playerPosition, playerForwardDirection, TagsAndLayers.Layers.PlayerMask, ref _enemyBuffer);
-            if (_hits > 1)
-                _hits = 1; //ensure we only damage first enemy
-            return true;
+            var hits = TryAttack(playerPosition, playerForwardDirection);
+
+            if (!hits.HasValue) return null;
```

Comment:
Instead of making this nullable, could we instead return -1 to signify a a failed `TryAttack()`? 

Personal preference, just find nullables a bit harder to read and like to avoid them.